count trees for start conditions like poor_hamlet

SimonHeimberg opened the pull request:
... instead of just checking for one tree.

**Type of change**
Enhancement

**Additional context**
Starting with poor hamlet and then realising there are not enough trees is demotivating and needs to start a new game.

Counting trees is easy, and each one produces one log. So do this. The minimum of logs needed is determined from waretable_trees. (This lists the ware you get if there are not enough trees.)
Exactly seen, the logs in the wares you get could be reduced by the number of trees. It is a harder start anyway if you have to grow the trees for getting logs. So do not reduce the logs.

The rocks are still not counted. But it is unlikely that there is only one rock around which has the size one. And counting rocks (with the size) is more difficult.

**To reproduce**

Steps to reproduce the behavior which cause the problem in master but not in your branch:
1. Select map "Atoll"
2. Choose to play at yellow position, select "poor hamlet" as starting condition (win condition does not matter)
3. Start the game
4. (You get the message that there are no rocks nearby but you get some wares instead.)
5. Problem: You only can build your first building. But you can not cut enough trees to build any other.
 
**New behavior**
...
4. You get two messages, one for no rocks, one for not enough trees.
5. With the additional wares, you can start your economy. (Of course you still must follow the correct order and never send a builder too early.)

SimonHeimberg committed:
count trees for start conditions like poor_hamlet

instead of just checking for one tree.

hessenfarmer reviewed (changes requested):
one thing inline.
As this is meant to be the bare minimum start I would like it better to reduce the wares you get for each tree present.

SimonHeimberg committed:
improve counting of trees

- stop counting when enough found (by using the new variable)
- reduce additional logs by the number of found trees
- adapt inbox message

hessenfarmer committed:
Merge branch 'master' into count_trees

SimonHeimberg committed:
fix counting of trees, adapt the right index

SimonHeimberg committed:
cleanup counting of trees

use consequently `waretable_trees.log`

hessenfarmer reviewed (approved):
forget about my preferences we have done it like this on various other locations.  so shall be good to go

hessenfarmer committed:
Merge branch 'master' into count_trees
